.github/workflows/ci-java.yml

```diff
@@ -98,26 +116,32 @@ jobs:
 
       - name: 'Run Maven Build'
         run: |
-          mvn --threads 1C \
+          mvn clean install \
+          --threads 1C \
           --no-transfer-progress \
-          -Dmaven.build.cache.enabled=false \
           -Dapidocgen.skip=false \
-          -Dmaven.jacoco.skip=false \
-          clean install
+          -Dmaven.build.cache.enabled=false \
+          -Dmaven.jacoco.skip=false
         working-directory: ./apps
 
       - name: 'Generate, Upload Build Reports'
+        if: ${{ github.event_type != 'merge_group' }}
+        env:
+          BFD_BRANCH: ${{ inputs.branch }}
         run: |
           mvn sonar:sonar \
-          -Dsonar.ci.autoconfig.disabled=true \
-          -Dsonar.projectKey="$SONAR_PROJECT_KEY" \
-          -Dsonar.host.url="$SONAR_HOST_URL" \
-          -Dsonar.token="$SONAR_TOKEN" \
-          -Dmaven.build.cache.enabled=false \
           -DskipITs \
           -DskipTests \
+          --no-transfer-progress \
+          -Dmaven.build.cache.enabled=false \
+          ${{ inputs.branch != null && '-Dsonar.branch.name=$BFD_BRANCH' || null }} -Dsonar.ci.autoconfig.disabled=true \
+          -Dsonar.coverage.jacoco.xmlReportPaths=target/coverage-reports/jacoco-it/jacoco.xml,target/coverage-reports/jacoco-ut/jacoco.xml \
           -Dsonar.exclusions='*.java' \
-          -Dsonar.coverage.jacoco.xmlReportPaths=target/coverage-reports/jacoco-it/jacoco.xml,target/coverage-reports/jacoco-ut/jacoco.xml
+          -Dsonar.host.url="$SONAR_HOST_URL" \
+          -Dsonar.projectKey="$SONAR_PROJECT_KEY" \
+          -Dsonar.projectVersion="$(git log -1 --format=%H)" \
```

Comment:
It's only a bit counter-intuitive, but, we need the output of the `git` binary here to set the appropriate sha. We'd love to be able to just refer to the built-in context and variable `github.sha`, but... that doesn't make sense for `workflow_dispatch` where it will always pick up on the `ref` but do the _wrong_ thing when the content of the `ref` and the input variable `branch` aren't the same.